app/src/main/java/ch/epfl/sweng/SDP/game/drawing/DrawingActivity.java

```diff
@@ -62,72 +40,18 @@
     private ImageView eraserButton;
     private ImageView bucketButton;
 
-    private final Database database = Database.INSTANCE;
-    private DatabaseReference timerRef;
-    private DatabaseReference stateRef;
-
-    protected final ValueEventListener listenerState = new ValueEventListener() {
-        @Override
-        public void onDataChange(@NonNull DataSnapshot dataSnapshot) {
-            Integer state = dataSnapshot.getValue(Integer.class);
-            if(state != null) {
-                GameStates stateEnum = GameStates.convertValueIntoState(state);
-                switch(stateEnum) {
-                    case START_VOTING_ACTIVITY:
-                        isVotingActivityLaunched = true;
-                        timerRef.removeEventListener(listenerTimer);
-                        Intent intent = new Intent(getApplicationContext(),
-                                VotingPageActivity.class);
-                        intent.putExtra("RoomID", roomID);
-                        startActivity(intent);
-                        break;
-                    default:
-                }
-            }
-        }
-
-        @Override
-        public void onCancelled(@NonNull DatabaseError databaseError) {
-            // Does nothing for the moment
-        }
-    };
-
-    protected final ValueEventListener listenerTimer = new ValueEventListener() {
-        @Override
-        public void onDataChange(@NonNull DataSnapshot dataSnapshot) {
-            Integer value = dataSnapshot.getValue(Integer.class);
-
-            if(value != null) {
-                ((TextView) findViewById(R.id.timeRemaining)).setText(String.valueOf(value));
-            }
-        }
-
-        @Override
-        public void onCancelled(@NonNull DatabaseError databaseError) {
-            // Does nothing for the moment
-        }
-    };
-
+    int getLayoutid() {
+        return R.layout.activity_drawing_offline;
+    }
```

Comment:
But is it useful? The resource id is used only once

Reply:
I created this function so I can inherit this value from in subclasses. It's a small "hack" used to subclass activities.

Reply:
But you @LudoHoff is right. I need to change the access rights to **protected** for this function

Reply:
But the layout of the subclasses will be different, hence you will still use it once no?

Reply:
No, Drawing Activity and Drawing game use the same layout

Reply:
What I mean is that DrawingGame for the offline mode and DrawingGameWithTimer for the online mode (which I would rename DrawingOffline and DrawingOnline by the way) have different layouts, and they are the subclassed activities we create. Since DrawingActivity is just the superclass, it will not be actually created and thus it does not need a layout: hence the method here is not useful.

Reply:
However, I just realised that I could remove the override in the DrawingOffline class

Reply:
Also change this to getLayoutId